src/MICore/MICommandFactory.cs

```diff
@@ -501,6 +501,34 @@ public virtual void DecodeExceptionReceivedProperties(Results miExceptionResult,
 
         #region Helpers
 
+        /// <summary>
+        /// Generates results that represent an emulated MI stopped record.
+        /// </summary>
+        public async Task<Results> GenerateStoppedRecordResults()
+        {
+            Results threadInfo = await this.ThreadInfo();
+            // Get the current thread identifier
+            string currentThreadId = threadInfo.FindString("current-thread-id");
+            Debug.Assert(!String.IsNullOrEmpty(currentThreadId), "Could not determine current thread.");
+            
+            ValueListValue threads = threadInfo.Find<ValueListValue>("threads");
+            Debug.Assert(null != threads, "Could not get thread information from debuggee.");
```

Comment:
I dug a little more to see what would actually happen if it couldn't fine something via the Find* methods. Those methods will throw MIResultFormatException. AD7Engine will handle any exception thrown during the call to ResumeFromLaunch, report the error, and stops debugging. The error message would read as "Unable to start debugging. Unrecognized format of field 'current-thread-id' in result: <pseudo-MI-record-string-for-thread-info-call>".